### PR TITLE
Add slugify fork that allows for dns safe slugging

### PR DESCRIPTION
**Pull request by agjohnson**
This adds a function that wraps the django core `slugify`, replacing multiple
non-alpha characters with `-`.

### CODE REVIEW

File: readthedocs/projects/models.py [hunk 1]
```diff
@@ -304,7 +303,7 @@ def save(self, *args, **kwargs):
         first_save = self.pk is None
         if not self.slug:
             # Subdomains can't have underscores in them.
-            self.slug = slugify(self.name).replace('_', '-')
+            self.slug = slugify(self.name)
```
Comment (ericholscher) [inline readthedocs/projects/models.py]: Shoudl this not be using dns_safe?

File: readthedocs/core/utils/__init__.py [hunk 2]
```diff
@@ -134,3 +138,18 @@ def send_email(recipient, subject, template, template_html, context=None,
...
         scheme='https', host=settings.PRODUCTION_DOMAIN)
     send_email_task.delay(recipient, subject, template, template_html, context)
+
+
+def slugify(value, *args, **kwargs):
+    """Add a DNS safe option to slugify
+
+    :param dns_safe: Remove underscores from slug as well
+    """
+    dns_safe = kwargs.pop('dns_safe', False)
```
Comment (ericholscher) [inline readthedocs/core/utils/__init__.py]: Feels like this should default to true. I generally think slugs shouldn't have _ and - in them, so happy to default it on.

Comment (agjohnson) [inline readthedocs/core/utils/__init__.py]: Yeah, this makes sense.  I was solidly on the fence and deferred the decision for later.

File: readthedocs/projects/models.py (hunk 1 above)
Comment (agjohnson) [inline readthedocs/projects/models.py]: Fixed with swapped logic

**Pull request merged**